front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -587,17 +615,27 @@ function Citations({
       // ref={citationContainer}
     >
       {activeReferences.map(({ document, index }) => {
-        const [documentCitation] = makeDocumentCitations([document]);
+        let citation;
+
+        if (isRetrievalDocumentType(document)) {
+          citation = makeDocumentCitations([document])[0];
+        } else if (isWebsearchResultType(document)) {
+          citation = makeWebsearchResultsCitations([document])[0];
+        }
+
+        if (!citation) {
```

Comment:
Same here if you replace the `else if` block from above you should not need this code.